backend/source/archive/models/google_takeout.py

```diff
@@ -16,13 +16,29 @@
 
 
 def e7_to_decimal(e7_coordinate: int) -> float:
-    return float(e7_coordinate) / 10000000
+    try:
+        return float(e7_coordinate) / 10000000
+    except:
+        return 
 
 
 def millis_str_to_time(timestamp: str) -> datetime:
     return datetime.fromtimestamp(int(timestamp) / 1000, tz=pytz.UTC)
 
 
+def tz_str_to_time(timestamp: str) -> datetime:
+    # if string is only millis
+    if timestamp.find('Z') == -1:
+        return millis_str_to_time(timestamp)
+
+    # Remove milliseconds if existing
+    ms_id = timestamp.find('.')
+    if not ms_id == -1:
+        timestamp = timestamp[:ms_id] + "Z"
+
+    return datetime.strptime(timestamp[:-1], '%Y-%m-%dT%H:%M:%S')
```

Comment:
Use `utils.json_to_datetime`. It handles potential issues with time zones, parses milliseconds if they're there, and is used across the code base.